hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutFromLoadAllOperation.java

```diff
@@ -70,8 +71,12 @@ public void run() throws Exception {
             Object previousValue = recordStore.putFromLoad(key, value);
 
             callAfterPutInterceptors(value);
+            // Null-pointer here, in case of null key loaded by MapLoader
+            checkNotNull(key, "Key loaded by a MapLoader cannot be null.");
```

Comment:
That is true, I mean it was like this before, but there's not reason we can't fix it now. So I added a guard for the interceptor, plus I moved the key validation higher to avoid de-serialization calls. The value validation has to stay where it is, otherwise it would be a change of behavior.